core/server/common/src/main/java/alluxio/master/journal/JournalTool.java

```diff
@@ -239,4 +194,299 @@ private static void usage() {
         "Read an Alluxio journal and write it to a directory in a human-readable format.", OPTIONS,
         "", true);
   }
+
+  /**
+   * An abstract class for journal dumpers.
+   */
+  abstract static class AbstractJournalDumper {
```

Comment:
The JournalTool class is quite long, should we separate these out to their own classes?